Duplicate definition in cpp codegen

```nim
import sequtils
import strutils


let rules = toSeq(lines("input"))
  .mapIt(it.split(" => ").mapIt(it.replace("/", "")))
  .mapIt((it[0], it[1]))
```

```
> nim cpp -d:release a21
Hint: used config file '/home/mstearn/nim/nim/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: a21 [Processing]
Hint: sequtils [Processing]
Hint: macros [Processing]
Hint: strutils [Processing]
Hint: parseutils [Processing]
Hint: math [Processing]
Hint: algorithm [Processing]
a21.nim(5, 5) Hint: 'rules' is declared but not used [XDeclaredButNotUsed]
CC: a21
CC: stdlib_system
CC: stdlib_sequtils
CC: stdlib_macros
CC: stdlib_strutils
CC: stdlib_parseutils
CC: stdlib_math
CC: stdlib_algorithm
Error: execution of an external compiler program 'g++ -c  -w -w -fpermissive -O3 -fno-strict-aliasing  -I/home/mstearn/nim/nim/lib -o /home/mstearn/aoc/21/nimcache/a21.o /home/mstearn/aoc/21/nimcache/a21.cpp' failed with exit code: 1

/home/mstearn/aoc/21/nimcache/a21.cpp:334:37: error: redefinition of ‘tySequence_sM4lkSb7zS6F7OVMvW9cffQ* result_Cbjk47dTBQHwrAZxXLxZZw_4’
 tySequence_sM4lkSb7zS6F7OVMvW9cffQ* result_Cbjk47dTBQHwrAZxXLxZZw_4;
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mstearn/aoc/21/nimcache/a21.cpp:321:37: note: ‘tySequence_sM4lkSb7zS6F7OVMvW9cffQ* result_Cbjk47dTBQHwrAZxXLxZZw_4’ previously declared here
 tySequence_sM4lkSb7zS6F7OVMvW9cffQ* result_Cbjk47dTBQHwrAZxXLxZZw_4;
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mstearn/aoc/21/nimcache/a21.cpp:335:16: error: redefinition of ‘NimStringDesc* x_D0OZ9aNVprOmety7qyz6CxQ_2’
 NimStringDesc* x_D0OZ9aNVprOmety7qyz6CxQ_2;
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mstearn/aoc/21/nimcache/a21.cpp:322:16: note: ‘NimStringDesc* x_D0OZ9aNVprOmety7qyz6CxQ_2’ previously declared here
 NimStringDesc* x_D0OZ9aNVprOmety7qyz6CxQ_2;
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mstearn/aoc/21/nimcache/a21.cpp:336:16: error: redefinition of ‘NimStringDesc* it_4in0LN2NNJ9bsya8DN5gD8A’
 NimStringDesc* it_4in0LN2NNJ9bsya8DN5gD8A;
                ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mstearn/aoc/21/nimcache/a21.cpp:323:16: note: ‘NimStringDesc* it_4in0LN2NNJ9bsya8DN5gD8A’ previously declared here
 NimStringDesc* it_4in0LN2NNJ9bsya8DN5gD8A;
                ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mstearn/aoc/21/nimcache/a21.cpp:340:16: error: redefinition of ‘NimStringDesc* it_4in0LN2NNJ9bsya8DN5gD8A_2’
 NimStringDesc* it_4in0LN2NNJ9bsya8DN5gD8A_2;
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mstearn/aoc/21/nimcache/a21.cpp:327:16: note: ‘NimStringDesc* it_4in0LN2NNJ9bsya8DN5gD8A_2’ previously declared here
 NimStringDesc* it_4in0LN2NNJ9bsya8DN5gD8A_2;
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Works fine with c codegen. This seems to be new in devel since it also works fine with 0.17.2

--- a/tests/cpp/t6986.nim
+++ b/tests/cpp/t6986.nim
@@ -1,0 +1,16 @@
+discard """
+  targets: "cpp"
+  action: "compile"
+"""
+
+import sequtils, strutils
+
+
+let rules = toSeq(lines("input"))
+  .mapIt(it.split(" => ").mapIt(it.replace("/", "")))
+  .mapIt((it[0], it[1]))
+
+
+proc pp(s: string): auto =
+  toSeq(lines(s)).mapIt(it.split(" => ").mapIt(it.replace("/", ""))).mapIt((it[0], it[1]))
+echo pp("input")

